Target data:

In [99]:
import pandas
target_df = pandas.read_csv('../data/model_data_2.csv')

In [100]:
target_df.head()

OWNER_CL_ID  ZIP_CD  DESIRED_CL_CCPC  DESIRED_CL_CCPCEC    sample  \
0    10004700   89011                0                  0   holdout   
1    10005cd7   92307                0                  0  training   
2    1000a8f6   78550                0                  0  training   
3    1000b62a   91789                1                  1  training   
4    1000cbcd   78635                0                  1  training   

        FILE_NAME  
0  file_89011.png  
1  file_92307.png  
2  file_78550.png  
3  file_91789.png  
4  file_78635.png

In [101]:
len(target_df)

250374

250374 different clients (unique clients). There a multiple policies (300K). 

Let's use the above dataset to split the pictures across the folder structure:

In [102]:
training_one_df = target_df[(target_df['DESIRED_CL_CCPCEC']  == 1) & (target_df['sample'] == 'training')]

In [103]:
training_one_df.head()

OWNER_CL_ID  ZIP_CD  DESIRED_CL_CCPC  DESIRED_CL_CCPCEC    sample  \
3     1000b62a   91789                1                  1  training   
4     1000cbcd   78635                0                  1  training   
9     1002a993    1886                0                  1  training   
13    10042a24   30643                0                  1  training   
30     10086ed   95121                0                  1  training   

         FILE_NAME  
3   file_91789.png  
4   file_78635.png  
9    file_1886.png  
13  file_30643.png  
30  file_95121.png

In [104]:
len(training_one_df)

35433

In [105]:
training_zero_df = target_df[(target_df['DESIRED_CL_CCPC']  == 0) & (target_df['sample'] == 'training')]

In [106]:
training_zero_df.head()

OWNER_CL_ID  ZIP_CD  DESIRED_CL_CCPC  DESIRED_CL_CCPCEC    sample  \
1    10005cd7   92307                0                  0  training   
2    1000a8f6   78550                0                  0  training   
4    1000cbcd   78635                0                  1  training   
5    10015f09   75010                0                  0  training   
9    1002a993    1886                0                  1  training   

        FILE_NAME  
1  file_92307.png  
2  file_78550.png  
4  file_78635.png  
5  file_75010.png  
9   file_1886.png

In [107]:
len(training_zero_df)

187952

In [108]:
12556 + 187952

200508

In [109]:
validation_one_df = target_df[(target_df['DESIRED_CL_CCPC']  == 1) & (target_df['sample'] == 'holdout')]

In [110]:
validation_zero_df = target_df[(target_df['DESIRED_CL_CCPC']  == 0) & (target_df['sample'] == 'holdout')]

In [111]:
training_one_df = training_one_df['FILE_NAME']

In [112]:
training_zero_df = training_zero_df['FILE_NAME']

In [113]:
validation_one_df = validation_one_df['FILE_NAME']

In [114]:
validation_zero_df = validation_zero_df['FILE_NAME']

In [115]:
training_one = training_one_df.tolist()

In [116]:
training_zero = training_zero_df.tolist()

In [117]:
validation_one = validation_one_df.tolist()

In [118]:
validation_zero = validation_zero_df.tolist()

In [119]:
'file_1886.png' in training_one

True

In [121]:
import os
from os import path
import shutil

src = "../imgs/"
dst = "../data/train/positive/"


files = [i for i in training_one]
for f in files:
    try:
        shutil.copy(src + str(f),dst)
    except:
        pass

In [123]:
import os
from os import path
import shutil

src = "../imgs/"
dst = "../data/train/negative/"


files = [i for i in training_zero]
for f in files:
    try:
        shutil.copy(src + str(f),dst)
    except:
        pass

In [124]:
import os
from os import path
import shutil

src = "../imgs/"
dst = "../data/validation/positive/"


files = [i for i in validation_one]
for f in files:
    try:
        shutil.copy(src + str(f),dst)
    except:
        pass

In [125]:
import os
from os import path
import shutil

src = "../imgs/"
dst = "../data/validation/negative/"


files = [i for i in validation_zero]
for f in files:
    try:
        shutil.copy(src + str(f),dst)
    except:
        pass

In [126]:
import os

path, dirs, files = next(os.walk("../data/train/positive/"))
file_count = len(files)
print(file_count)
print(len(training_one_df))

8735
35433


In [127]:
path, dirs, files = next(os.walk("../data/train/negative/"))
file_count = len(files)
print(file_count)
print(len(training_zero_df))

16139
187952


In [128]:
path, dirs, files = next(os.walk("../data/validation/positive/"))
file_count = len(files)
print(file_count)
print(len(validation_one_df))

2065
3123


In [129]:
path, dirs, files = next(os.walk("../data/validation/negative/"))
file_count = len(files)
print(file_count)
print(len(validation_zero_df))

10482
46743


In [132]:
8735 + 16139 + 2065 + 10482

37421

In [131]:
12556 + 187952 + 3123 + 46743

250374

In [98]:
import os

path, dirs, files = next(os.walk("../imgs/"))
file_count = len(files)
print(file_count)

23010
